_posts/2024-02-07-#28950.md

```diff
@@ -8,16 +8,56 @@ authors: [instagibbs]
+
+- `maxburnamount` is a similar optional parameter for the `sendrawtransaction` RPC that helps prevent users from accidentally burning funds. Transactions with a data carrier `OP_RETURN` output amount that exceeds `maxburnamount` (default value: 0) are not submitted or broadcasted.
+
+- Upon decoding a transaction in `sendrawtransaction` and `testmempoolaccept` RPCs, `maxburnamount` sanity check is immediately performed.
+
+- In the `sendrawtransaction` RPC, `maxfeerate` sanity check is performed by calculating the transaction size fee at `maxfeerate` fee rate (as [`max_raw_tx_fee`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L89)) which is passed to [`BroadcastTransaction`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/node/transaction.cpp#L33), transactions whose base fee exceeds the `max_raw_tx_fee` are neither broadcasted nor added to the mempool.
+
+- `testmempoolaccept` performs `maxfeerate` sanity check after package processing. It ensures that the individual base fee of all the successfully validated transactions [does not exceed](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L218) the fee of the transaction size at `maxfeerate` fee rate.
+
+- This PR adds optional parameters `maxfeerate` and `maxburnamount` to the [`submitpackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L816) RPC, enabling similar sanity checks on package transactions.
+
+- After [decoding each package transaction in `submitpackage`](https://github.com/bitcoin/bitcoin/blob/e69796c79c0aa202087a13ba62d9fbcc1c8754d4/src/rpc/mempool.cpp#L814), the transaction outputs are subjected to a `maxburnamount` sanity check.
+
+- The PR introduces the `max_sane_feerate` parameter to the [`ProcessNewPackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.h#L282) function. `submitpackage` now passes the value of `maxfeerate` as the `max_sane_feerate` when calling `ProcessNewPackage`.
+
+- `ProcessNewPackage` is updated to forward the `max_sane_feerate` to [`AcceptPackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.cpp#L1450), which performs the sanity check during [`Sub package`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/validation.cpp#L1425) evaluation.
```

Comment:
It seems odd to point to `AcceptSubPackage` instead of the line in which the max feerate is checked?

Reply:
t is checked in two different places: when the sub-package is a single transaction and when it has multiple transactions in `AcceptSingleTransaction` and `AcceptMultipleTransactions`.

We determine which one to call in `AcceptSubPackage` and perform the actual check within that scope. 
I thought this could easily be seen after going through the code within the `AcceptSubPackage` function?

Reply:
Of course I will be nice if you could help in making this clearer, you know better :D 
Thank you